resources/clients/react/elwing/src/plugins/item-tracker/work-item/WorkItemListComponent.tsx

```diff
@@ -0,0 +1,179 @@
+/**
+ * An element that displays a list of work items that are retrieved from a REST service.
+ *
+ * - Select Active or Archived to display work items with the specified state.
+ * - Select the wastebasket icon to archive an active item.
+ * - Select 'Add item' to add a new item.
+ * - Enter a recipient email and select 'Send report' to send a report of work items.
+ */
+
+import { useEffect, useState } from "react";
+import { WorkItem, WorkItemStatus } from "./WorkItemService";
+import { workItemService as service } from "./WorkItemService";
+import {
+  Alert,
+  Button,
+  FormField,
+  Input,
+  Select,
+  SpaceBetween,
+  Table,
+} from "@cloudscape-design/components";
+import { OptionDefinition } from "@cloudscape-design/components/internal/components/option/interfaces";
+import { useItemTrackerAction, useItemTrackerState } from "../ItemTrackerStore";
+
+const STATUS_OPTIONS: OptionDefinition[] = [
+  { value: "active", label: "Active" },
+  { value: "archived", label: "Archived" },
+  { value: "", label: "All" },
+];
+
+export const WorkItemControls = ({
+  sendReport,
+}: {
+  sendReport: (email: string) => Promise<void>;
+}) => {
+  const [email, setEmail] = useState("");
+  const [status, setStatus] = useState(STATUS_OPTIONS[0]);
+  const { setStatus: handleStatusChange } = useItemTrackerAction();
+
+  useEffect(() => {
+    handleStatusChange((status.value as WorkItemStatus) ?? "");
+  }, [status, handleStatusChange]);
+
+  return (
+    <SpaceBetween size="s">
+      <FormField>
+        <Select
+          selectedOption={status}
+          onChange={(event) => setStatus(event.detail.selectedOption)}
+          options={STATUS_OPTIONS}
+        />
+      </FormField>
+      <FormField
+        label="Email Report"
+        description="You must first register the recipient's email with Amazon SES."
+      >
+        <Input
+          value={email}
+          onChange={(event) => setEmail(event.detail.value)}
+          placeholder="Recipient's email"
+        />
+      </FormField>
+      <Button disabled={email === ""} onClick={() => sendReport(email)}>
```

Comment:
Could possibly be shortened to `!email`. No strong feelings either way.

Reply:
.isEmpty(), please, TC39. Generally, I eschew the conversion operators in JS.

Reply:
Fair enough.